### PR TITLE
feat(mute-metric-alerts): Add mute button to metric alerts page

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr adds in the mute button and banner showing mute status to the metric alerts page.

### CODE REVIEW

File: static/app/views/alerts/rules/utils.tsx [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import IdBadge from 'sentry/components/idBadge';
 import {t} from 'sentry/locale';
 import {Organization, Project} from 'sentry/types';
+import type {IssueAlertRule} from 'sentry/types/alerts';
+import {RuleActionsCategories} from 'sentry/types/alerts';
```
Comment (scttcper) [inline static/app/views/alerts/rules/utils.tsx:5]: just combine these

File: static/app/views/alerts/rules/metric/details/body.tsx [hunk 2]
```diff
@@ -172,6 +177,22 @@ export default class DetailsBody extends Component<Props> {
...
+                      "[creator] muted this alert so these notifications won't be sent in the future.",
+                      {creator: rule.snoozeCreatedBy}
+                    )
+                  : tct(
+                      "[creator] muted this alert[forEveryone]so you won't get these notifications in the future.",
+                      {
+                        creator: rule.snoozeCreatedBy,
+                        forEveryone: rule.snoozeForEveryone ? ' for everyone ' : ' ',
+                      }
+                    )}
```
Comment (scttcper) [inline static/app/views/alerts/rules/metric/details/body.tsx:193]: dumb question but wouldn't metric alerts always be for everyone or are we looking for specific email my team -> this user is muted

Comment (ceorourke) [reply static/app/views/alerts/rules/metric/details/body.tsx:193]: Screenshot 2023-06-15 at 12 31 25 PM

you can set it to email a user or yeah if it's a team it emails each user in the team and one person can mute it for themselves.

**Review by scttcper — APPROVED** [review]

File: static/app/views/alerts/rules/metric/details/index.spec.tsx [hunk 3]
```diff
@@ -129,4 +129,56 @@ describe('MetricAlertDetails', () => {
     expect(incidentMock).toHaveBeenCalled();
     expect(issuesRequest).toHaveBeenCalled();
   });
+
+  it('renders mute button for metric alert', async () => {
```
Comment (malwilley) [inline static/app/views/alerts/rules/metric/details/index.spec.tsx:133]: Good test! Thanks for adding 👍

File: static/app/views/alerts/rules/metric/details/header.tsx [hunk 4]
```diff
@@ -39,6 +53,9 @@ function DetailsHeader({hasMetricRuleDetailsError, rule, organization, project}:
     },
   };
 
+  const hasSnoozeFeature = organization.features.includes('mute-metric-alerts');
+  const isSnoozed = rule?.snooze || false;
```
Comment (malwilley) [inline static/app/views/alerts/rules/metric/details/header.tsx:57]: very small nit: prefer `??` to `||` but it doesn't matter here

**Review by malwilley — APPROVED** [review]

**Pull request merged**